doc/guides/west/manifest.rst

```diff
@@ -1179,6 +1189,437 @@ processed in this order:
+         groups:
+           - groupA
+       - name: baz
+         groups:
+           - groupA
+           - groupB
+
+     defaults:
+       remote: example-remote
+     remotes:
+       - name: example-remote
+         url-base: https://git.example.com
+
+The ``manifest.groups`` configuration option is set to ``-groupA,-groupB``
+(you can ensure this by running ``west config manifest.groups --
+"-groupA,-groupB"``).
+
+In this case, both ``groupA`` and ``groupB`` are blocked.
+
+Therefore, projects ``foo`` and ``bar`` are both inactive.
+
+.. _west-manifest-submodules:
+
+Git Submodules in Projects
+**************************
+
+You can use the ``submodules`` keys briefly described :ref:`above
+<west-manifest-files>` to force ``west update`` to also handle any `Git
+submodules`_ configured in project's git repository. The ``submodules`` key can
+appear inside ``projects``, like this:
+
+.. code-block:: YAML
+
+   manifest:
+     projects:
+       - name: some-project
+         submodules: ...
+
+The ``submodules`` key can be a boolean or a list of mappings. We'll describe
+these in order.
+
+Option 1: Boolean
+=================
+
+This is the easiest way to use ``submodules``.
+
+If ``submodules`` is ``true`` as a ``projects`` attribute, ``west update`` will
+recursively update the project's Git submodules whenever it updates the project
+itself. If it's ``false`` or missing, it has no effect.
+
+For example, let's say you have a source code repository ``foo``, which has
+some submodules, and you want ``west update`` to keep all of them them in sync,
+along with another project named ``bar`` in the same workspace.
+
+You can do that with this manifest file:
+
+.. code-block:: yaml
+
+   manifest:
+     projects:
+       - name: foo
+         submodules: true
+       - name: bar
+
+Here, ``west update`` will initialize and update all submodules in ``foo``. If
+``bar`` has any submodules, they are ignored, because ``bar`` does not have a
+``submodules`` value.
+
+Option 2: List of mappings
+==========================
+
+The ``submodules`` key may be a list of mappings. Each element in the list
+defines the name of the submodule to update, and the path -- relative to the
+project's absolute path in the workspace -- to use for the submodule.
+
+For example, let's say you have a source code repository ``foo``, which has
+many submodules, and you want ``west update`` to keep some but not all of them
+in sync, along with another project named ``bar`` in the same workspace.
+
+You can do that with this manifest file:
+
+.. code-block:: yaml
+
+   manifest:
+     projects:
+       - name: foo
+         submodules:
+           - name: foo-first-sub
+             path: path/to/foo-first-sub
+           - name: foo-second-sub
+             path: path/to/foo-second-sub
+       - name: bar
+
+Here, ``west update`` will initialize and update just the submodules in ``foo``
+with paths ``path/to/foo-first-sub`` and ``path/to/foo-second-sub``. Any
```

Comment:
Based on this comment https://github.com/zephyrproject-rtos/zephyr/pull/31241/files#r557368457, I think a very important piece of information is needed here:
```suggestion
Here, ``west update`` will initialize and update just the submodules in ``foo``
with paths ``path/to/foo-first-sub`` and ``path/to/foo-second-sub``.  ``path/to/foo-first-sub`` and ``path/to/foo-second-sub`` will also have their submodules updated recursively. Any
```